tests/Unit/DataStructures/Test_Mesh.cpp

```diff
@@ -14,99 +14,126 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Mesh", "[DataStructures][Unit]") {
+                                      {Spectral::Quadrature::Gauss,
+                                       Spectral::Quadrature::GaussLobatto}});
+    CHECK(mesh2d.quadrature(0) == Spectral::Quadrature::Gauss);
+    CHECK(mesh2d.quadrature(1) == Spectral::Quadrature::GaussLobatto);
     const auto mesh2d_sliced_0 = mesh2d.slice_away(0);
     CHECK(mesh2d_sliced_0.number_of_grid_points() == 3);
-    CHECK(mesh2d_sliced_0.bases()[0] == Spectral::Basis::Legendre);
-    CHECK(mesh2d_sliced_0.quadratures()[0] ==
-          Spectral::Quadrature::GaussLobatto);
+    CHECK(mesh2d_sliced_0.bases() ==
+          std::array<Spectral::Basis, 1>{{Spectral::Basis::Legendre}});
+    CHECK(mesh2d_sliced_0.quadratures() ==
+          std::array<Spectral::Quadrature, 1>{
+              {Spectral::Quadrature::GaussLobatto}});
     const auto mesh2d_sliced_1 = mesh2d.slice_away(1);
     CHECK(mesh2d_sliced_1.number_of_grid_points() == 2);
-    CHECK(mesh2d_sliced_1.bases()[0] == Spectral::Basis::Legendre);
-    CHECK(mesh2d_sliced_1.quadratures()[0] == Spectral::Quadrature::Gauss);
+    CHECK(mesh2d_sliced_1.bases() ==
+          std::array<Spectral::Basis, 1>{{Spectral::Basis::Legendre}});
+    CHECK(mesh2d_sliced_1.quadratures() ==
+          std::array<Spectral::Quadrature, 1>{{Spectral::Quadrature::Gauss}});
     const Mesh<3> mesh3d{
         {{2, 3, 4}},
         {{Spectral::Basis::Legendre, Spectral::Basis::Legendre}},
         {{Spectral::Quadrature::GaussLobatto, Spectral::Quadrature::Gauss,
           Spectral::Quadrature::GaussLobatto}}};
-    CHECK(mesh3d[0] == 2);
-    CHECK(mesh3d[1] == 3);
-    CHECK(mesh3d[2] == 4);
+    CHECK(mesh3d.extents() == Index<3>{2, 3, 4});
+    CHECK(mesh3d.extents(0) == 2);
+    CHECK(mesh3d.extents(1) == 3);
+    CHECK(mesh3d.extents(2) == 4);
     CHECK(mesh3d.number_of_grid_points() == 24);
-    CHECK(mesh3d.bases()[0] == Spectral::Basis::Legendre);
-    CHECK(mesh3d.quadratures()[0] == Spectral::Quadrature::GaussLobatto);
-    CHECK(mesh3d.bases()[1] == Spectral::Basis::Legendre);
-    CHECK(mesh3d.quadratures()[1] == Spectral::Quadrature::Gauss);
-    CHECK(mesh3d.bases()[2] == Spectral::Basis::Legendre);
-    CHECK(mesh3d.quadratures()[2] == Spectral::Quadrature::GaussLobatto);
+    CHECK(mesh3d.bases() ==
+          std::array<Spectral::Basis, 3>{{Spectral::Basis::Legendre,
+                                          Spectral::Basis::Legendre,
+                                          Spectral::Basis::Legendre}});
+    CHECK(mesh3d.quadratures() ==
+          std::array<Spectral::Quadrature, 3>{
+              {Spectral::Quadrature::GaussLobatto, Spectral::Quadrature::Gauss,
+               Spectral::Quadrature::GaussLobatto}});
+    CHECK(mesh3d.basis(0) == Spectral::Basis::Legendre);
+    CHECK(mesh3d.quadrature(0) == Spectral::Quadrature::GaussLobatto);
+    CHECK(mesh3d.basis(1) == Spectral::Basis::Legendre);
+    CHECK(mesh3d.quadrature(1) == Spectral::Quadrature::Gauss);
```

Comment:
see if you can make a helper function that loops over the individual access case as well as checks the array access case